### PR TITLE
Hook up Sleef double intrinsics

### PR DESCRIPTION
**Pull request by resistor**

### CODE REVIEW

**Comment by protonu** [comment]
I was wondering if there are tests for this?

**Comment by resistor** [comment]
@protonu Tests added.

File: test/cpp/tensorexpr/test_llvm.cpp [hunk 1]
```diff
@@ -311,30 +311,76 @@ void testLLVMVecLoadStoreTest() {
     float ref = std::Name(0.5f);                                 \
     EXPECT_EQ(cg.value<int>(args), 0);                           \
     for (int i = 0; i < Lanes; i++) {                            \
-      EXPECT_EQ(a_buffer[i], val);                               \
+      EXPECT_FLOAT_EQ(a_buffer[i], val);                               \
```
Comment (protonu) [inline test/cpp/tensorexpr/test_llvm.cpp:314]: This is my fault, this test was incomplete, but it worked for SSE. This is missing:
    for (int i = 0; i < Lanes; i++) {                            \
      EXPECT_EQ(b_buffer[i], ref);                               \	      EXPECT_FLOAT_EQ(a_buffer[i], val);                               \
    }                                                            \	    }

Comment (protonu) [reply test/cpp/tensorexpr/test_llvm.cpp:314]: Do you think it's okay to get this, and then debug this?
I also have double precision version. In it I modified the test to call sleef ( I need to make a small modification to the cmake for tests). That showed errors in DP

Comment (resistor) [reply test/cpp/tensorexpr/test_llvm.cpp:314]: I'm not worried about calling Sleef.  I verified that PyTorch eager mode is using these Sleef functions, so any inaccuracies are already in the baseline.

**Review by protonu — APPROVED** [review]

**Pull request merged**